enterprise/views.py

```diff
@@ -381,9 +392,39 @@ def post_course_specific_consent(self, request, course_id, consent_provided):
                     'consent_granted': consent_provided,
                 }
             )
+
         if not consent_provided:
-            failure_url = request.POST.get('failure_url') or reverse('dashboard')
             return redirect(failure_url)
+
+        # Check if the leaner has come from enterprise course enrollment page
```

Comment:
@haikuginger you are suggesting to create a new GET class view, e.g. `DataSharingConsentEnrollment` which will accept the `course id` and `course mode` and after handle the enrollment or redirection to ecommerce, right?